docs/DeveloperGuide.md

```diff
@@ -154,6 +161,76 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Schedule feature
+
+Users can view a calendar that contains all existing lessons to visualise their schedule and plan ahead.
+
+#### Current Implementation
+
+TAB uses the [CalendarFX](https://dlsc.com/products/calendarfx/) library to implement its schedule view.
+Each lesson stored in the `Model` component is also converted to a CalendarFX `Entry` and maintained in the `CalendarEntryList` of `AddressBook`. 
+The `Ui` component shows this list of entries in the `SchedulePanel` using CalendarFX's `DetailedWeekView`.
+
+The following code snippet shows how `SchedulePanel` is initialised from the `Model` component's `Calendar`.
+
+```java
+public class SchedulePanel extends UiPart<Region> {
+
+    private static final String FXML = "SchedulePanel.fxml";
+
+    private final Logger logger = LogsCenter.getLogger(this.getClass());
+
+    private final DetailedWeekView calendarView;
+
+    @FXML
+    private StackPane scheduleView;
+
+    /**
+     * Creates a {@code SchedulePanel} with the given {@code Calendar}.
+     */
+    public SchedulePanel(Calendar calendar) {
+        super(FXML);
+        calendarView = new DetailedWeekView();
+        initialiseCalendar(calendar);
+        createTimeThread();
+    }
+
+    /**
+     * Sets up CalendarFX.
+     * Adapted from CalendarFX developer manual.
+     * http://dlsc.com/wp-content/html/calendarfx/manual.html#_quick_start
+     */
+    public void initialiseCalendar(Calendar calendar) {
+        CalendarSource calendarSource = new CalendarSource();
+        calendarSource.getCalendars().addAll(calendar);
+        calendarView.getCalendarSources().addAll(calendarSource);
+        
+        calendarView.setStartTime(TimeRange.DAY_START);
+        calendarView.setEndTime(TimeRange.DAY_END);
+        // other settings...
+        
+    }
+    
+    // other SchedulePanel methods...
+}
```

Comment:
Can give a simplified version of the code, e.g. remove the "adapted from" can also add comments to explain what the things mean `private final Logger logger = LogsCenter.getLogger(this.getClass());` can write a comment at the side to explain so the code snippet is clearer